Sources/Rendering/Core/ImageArrayMapper/index.js

```diff
@@ -0,0 +1,171 @@
+    if (!inputCollection || inputCollection.empty()) {
+      return;
+    }
+
+    const perImageMap = inputCollection.map((image, index) => {
+      const dim = image.getDimensions();
+      const out = new Array(dim[model.slicingMode]);
+      for (let i = 0; i < out.length; ++i) {
+        out[i] = { imageNumber: index, subSlice: i };
+      }
+      return out;
+    });
+
+    model.sliceToSubSliceMap = perImageMap.flat();
+    publicAPI.modified();
+  };
+
+  const _superSetInputData = publicAPI.setInputData;
+
+  //------------------
+  // Public functions
+
+  publicAPI.setInputData = (inputData) => {
+    _superSetInputData(inputData);
+    _computeSliceToSubSliceMap();
+  };
+
+  // Overrides vtkAbstractImageMapper.getImage()
+  publicAPI.getImage = (slice = publicAPI.getSlice() ?? 0) => {
+    const imageCollection = publicAPI.getInputData();
+    if (!imageCollection) {
+      vtkWarningMacro(`No input set.`);
+    } else if (slice < 0 || slice >= publicAPI.getTotalSlices()) {
+      vtkWarningMacro(`Invalid slice number.`);
+    } else {
+      return imageCollection.getItem(
+        model.sliceToSubSliceMap[slice].imageNumber
+      );
+    }
+    return null;
+  };
+
+  publicAPI.getBounds = () => {
+    const image = publicAPI.getCurrentImage();
+    if (!image) {
+      return vtkMath.createUninitializedBounds();
+    }
+
+    const inputCollection = publicAPI.getInputData();
+    const bounds = [...vtkBoundingBox.INIT_BOUNDS];
+    inputCollection.forEach((element) => {
+      const sb = element.getBounds();
+      vtkBoundingBox.addBounds(
+        bounds,
+        sb[0],
+        sb[1],
+        sb[2],
+        sb[3],
+        sb[4],
+        sb[5]
+      );
+    });
+    return bounds;
+  };
+
+  publicAPI.getClosestIJKAxis = () => ({
+    ijkMode: model.slicingMode,
+    flip: false,
+  });
+
+  publicAPI.computeTotalSlices = () => {
+    const inputCollection = publicAPI.getInputData();
+    return inputCollection.reduce(
+      (accumulator, currImage) =>
+        currImage.getDimensions()[model.slicingMode] + accumulator,
+      0
+    );
+  };
+
+  publicAPI.getTotalSlices = () => model.sliceToSubSliceMap.length;
+
+  // set slice number in terms of imageNumber and subSlice number.
+  publicAPI.setSlice = (slice) => {
+    const inputCollection = publicAPI.getInputData();
+    if (
+      publicAPI.getTotalSlices() === 0 ||
+      inputCollection.getMTime() > publicAPI.getMTime()
+    ) {
+      _computeSliceToSubSliceMap();
+    }
+
+    // const totalSlices = publicAPI.computeTotalSlices();
+    const totalSlices = publicAPI.getTotalSlices();
+    if (slice >= 0 && slice <= totalSlices) {
+      model.slice = slice;
+      publicAPI.modified();
+    } else {
+      vtkErrorMacro(
+        `Slice number out of range. Acceptable range is: [0, ${totalSlices}]`
+      );
+    }
+  };
+
+  publicAPI.computeSlice = (imageNumber, subSlice) =>
+    model.sliceToSubSliceMap.findIndex((x) => x === { imageNumber, subSlice });
```

Comment:
shouldn't it be a deepEqual instead ?

Reply:
Or at least write out the comparison: `x.imageNumber === imageNumber && x.subSlice === subSlice`. As it's written, it will always eval to false.